src/core/shape_inference/include/transpose_shape_inference.hpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <openvino/core/validation_util.hpp>
+#include <openvino/op/transpose.hpp>
+
+#include "utils.hpp"
+namespace ov {
+namespace op {
+namespace v1 {
+template <typename Rank>
+bool is_valid_permutation(const ov::AxisVector& permutation, const Rank& rank) {
```

Comment:
Can we reuse `ngraph::apply_permutation()` and `ngraph::is_valid_permutation()` from `openvino/src/core/src/util.cpp`?

Reply:
The problem here is that the shape infer function cannot reference the header file from ngraph part

Reply:
Transpose shape inference is created in master, so just revert my changes.